includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+				// All keys in the database and not in blueprint are not included
+				if (isset($flow_data) && array_key_exists($key, $flow_data) && !is_array($value)) {
+					// Updates any default value added to an Existing Key in the blueprint into the database
+					if(isset($value) && empty($flow_data[$key])) 
+						$updated_flow_data[$key] = $value;
+
+					// Retains the user entered value at the time of onboarding
+					else
+						$updated_flow_data[$key] = $flow_data[$key];
+				}
+
+				// Adds new key-value pair added to the blueprint into the database
+				else
+					$updated_flow_data[$key] = $value;
+
+				if ( is_array( $value ) ) {
+
+					// For Keys having Empty Arrays. Eg: isViewed, other
+					if(empty($value)) {
+						if(empty($flow_data[$key]))
+							$updated_flow_data[$key] = $value;
+						else
+							$updated_flow_data[$key] = $flow_data[$key];
+					}
+
+					// To handle Indexed Arrays gracefully
+					if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+
+						// To check if an Indexed Array is further Nested or Not
+						foreach($value as $index_key => $index_value) {
+
+							// For Indexed Arrays having values as an Array: Indexed/Associative
+							if(is_array($index_value)) {
+								if(empty($flow_data[$key][$index_key]))
+									$updated_flow_data[$key][$index_key] = $index_value;
+								else
+									$updated_flow_data[$key][$index_key] = $flow_data[$key][$index_key];
+							}
+
+							// For Indexed Arrays having values as a String/Boolean
+							else {
+								if(empty($flow_data[$key]))
+									$updated_flow_data[$key] = $value;
+								else
+									$updated_flow_data[$key] = $flow_data[$key];
+							}
+						}
+					}
+					else
+						$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key], $updated_flow_data[$key]);
+				}
+			}
+			return $updated_flow_data;
+		  }
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	public static function update_post_call_data_recursive(&$flow_data, &$updated_flow_data, &$params, &$flag = '') {
+		{
+			foreach ($flow_data as $key => $value)
+			{ 
+				if(is_array($params[$key]) && !is_array($value)) {
+					$flag = $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value);
+					break;
+				}
+
+				// Updates value entered by the user
+				if (isset($params[$key]) && array_key_exists($key, $params)) {
+					if(strcmp(gettype($value), gettype($params[$key])) === 0) 
+						$updated_flow_data[$key] = $params[$key];
+					else {
+						$flag = $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value);
+						break;
+					}
+				}
+
+				elseif(isset($params) && !array_key_exists($key, $params)) {
+					$updated_flow_data[$key] = $flow_data[$key];
+				}
+
+				// Retains the Blueprint Value if no input from the User
+				else $updated_flow_data[$key] = $value;
```

Comment:
```suggestion
				// Retains the DB Value if no input from the User
				else $updated_flow_data[$key] = $value;
```
I might be wrong here but as far as I can see in relation to by previous comment, if the array_key does not exist in params or is not set or both, then we can just default to the value from the DB, don't know why we are checking `isset($params)` and then using `$flow_data[$key]` which is nothing but `$value`.